### DISCUSSION TITLE
Are there any plans to rewrite the project in rust?

### DISCUSSION THREAD

**Discussion by RustLoverTheCoder** [wundergraph/cosmo#877] [General]
Are there any plans to rewrite the project in rust? I think the advantage of the apollo router is that it uses rust

**Comment by StarpTech** [comment]
Hello @RustLoverTheCoder,

Currently, there are no plans to rewrite the project in Rust. Our primary focus is on enhancing the existing codebase, improving performance, and adding new features. While Rust is known for its safety and performance benefits, a complete rewrite would require significant resources and time, which isn't justifiable given our current priorities.

Moreover, a GraphQL gateway involves much more than just the programming language. You can check our [blog](https://wundergraph.com/blog) to get a better impression. In terms of performance, we can achieve comparable or even superior results to our competitors. At WunderGraph, we are Go experts. In our opinion, Go is the ideal language for building a reliable and efficient gateway, while also maintaining high productivity.

**Reply by RustLoverTheCoder** [reply]
Ok, thank you for your answer
